### PR TITLE
`tools/importer-rest-api-specs`: removing the logic for outputting V1 of the API Definitions

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR removes the logic which output the V1 API Definitions - and also removes the feature toggle for V2, since this is now defaulted on.

I've also updated the README to account for the changes - and called out the paths which will be removed in the near future.

### CODE REVIEW

File: tools/importer-rest-api-specs/pipeline/run_importer.go [hunk 1]
```diff
@@ -40,20 +39,10 @@ func runImporter(input RunInput, generationData []discovery.ServiceInput, swagge
...
-		if featureflags.GenerateV2APIDefinitions {
-			serviceDirectoryV2 := path.Join(input.OutputDirectoryJson, serviceName)
-			logger.Debug("recreating the V2 working directory at %q for Service %q", serviceDirectoryV2, serviceName)
-			if err := dataapigenerator.RecreateDirectory(serviceDirectoryV2, logger); err != nil {
-				fmt.Errorf("recreating JSON directory %q for service %q", serviceDirectoryV2, serviceName)
-			}
+		serviceDirectoryV2 := path.Join(input.OutputDirectory, serviceName)
+		logger.Debug("recreating the V2 working directory at %q for Service %q", serviceDirectoryV2, serviceName)
+		if err := dataapigeneratorjson.RecreateDirectory(serviceDirectoryV2, logger); err != nil {
+			fmt.Errorf("recreating JSON directory %q for service %q", serviceDirectoryV2, serviceName)
```
Comment (stephybun) [inline tools/importer-rest-api-specs/pipeline/run_importer.go:45]:
```suggestion
		serviceDirectory := path.Join(input.OutputDirectory, serviceName)
		logger.Debug("recreating the working directory at %q for Service %q", serviceDirectory, serviceName)
		if err := dataapigeneratorjson.RecreateDirectory(serviceDirectory, logger); err != nil {
			fmt.Errorf("recreating directory %q for service %q", serviceDirectory, serviceName)
```

File: tools/importer-rest-api-specs/pipeline/task_generate_data_api_definitions_v2.go
Comment (stephybun) [inline tools/importer-rest-api-specs/pipeline/task_generate_data_api_definitions_v2.go:1]: Rename this file: `task_generate_data_api_definitions_v2.go` -> `task_generate_data_api_definitions.go`

File: tools/importer-rest-api-specs/pipeline/task_output_metadata.go [hunk 2]
```diff
@@ -3,24 +3,14 @@ package pipeline
...
-		// This is named metadata since it could make sense to output additional items in the future
-		// such as the licence for this imported data etc
-		if err := dataapigeneratorjson.OutputMetaData(input.OutputDirectoryJson, swaggerGitSha); err != nil {
-			return fmt.Errorf("outputting the Revision ID into the V2 Data: %+v", err)
-		}
+func (pipelineTask) outputMetaData(input RunInput, swaggerGitSha string) error {
+	// This is named metadata since it could make sense to output additional items in the future
+	// such as the licence for this imported data etc
+	if err := dataapigeneratorjson.OutputMetaData(input.OutputDirectory, swaggerGitSha); err != nil {
+		return fmt.Errorf("outputting the Revision ID into the V2 Data: %+v", err)
```
Comment (stephybun) [inline tools/importer-rest-api-specs/pipeline/task_output_metadata.go:13]:
```suggestion
		return fmt.Errorf("outputting the Revision ID: %+v", err)
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @tombuildsstuff. Looks mostly fine, a few minor renames are missing. Can we also remove the steps where we build the legacy Data API in this GHA
https://github.com/hashicorp/pandora/blob/8a17f3073483253f71007d0350ab920ce7fd66e3/.github/workflows/automation-regenerate-terraform.yaml#L27-L35

File: .github/workflows/automation-regenerate-terraform.yaml [hunk 3]
```diff
@@ -29,11 +29,6 @@ jobs:
         with:
           dotnet-version: 7.0.x
```
Comment (stephybun) [inline .github/workflows/automation-regenerate-terraform.yaml:30]: Shouldn't need this step anymore either?

**Review by stephybun — APPROVED** [review]
There's still a GHA step that I think can be removed, otherwise LGTM 👍

**Comment by tombuildsstuff** [comment]
The Graph failure is unrelated / fine to skip for now.

**Pull request merged**